cognite/client/_api/datapoints.py

```diff
@@ -786,15 +787,20 @@ def _get_windows(self, id, start, end, granularity, request_limit, user_limit):
             "1d"
         ) < cognite.client.utils._time.granularity_to_ms(granularity):
             count_granularity = granularity
-        res = self._get_datapoints_with_paging(
-            start=start,
-            end=end,
-            ts_item={"id": id},
-            aggregates=["count"],
-            granularity=count_granularity,
-            include_outside_points=False,
-            limit=None,
-        )
+        try:
+            res = self._get_datapoints_with_paging(
+                start=start,
+                end=end,
+                ts_item={"id": id},
+                aggregates=["count"],
+                granularity=count_granularity,
+                include_outside_points=False,
+                limit=None,
+            )
+        except CogniteAPIError:
```

Comment:
Could we be more specific about what type of CogniteAPIError we catch here?

Reply:
not really. in a sense failure here already gives some specifics in that (1) a normal datapoint request succeeded and (2) the type of error is not worth retrying, or retries failed.
we could check the text or error code after catching, but in any case, the user didn't ask us to get count aggregate, it's not strictly needed, and our response is to simply get datapoints with a different (non-paralellized) strategy, so we should avoid passing the exception up to the user.

Reply:
It could fail for other interesting reasons, which could be relevant. Not sure how we feel about logging but it might be worth doing to not hide this completely. On the other hand we don't seem to have a logger available so maybe that's not the way, @erlendvollset would know. :)

Since we're doing `if not res:` immediately after it seems this could be simplified to
```
        except CogniteAPIError:
            # assume it's a string based series or aggregates not yet calculated
            return [_DPWindow(start, end)]
```

Reply:
@wjoel for floating point timeseries with heavy ingestion / out of date aggregates we could have res=[] without exception (see issue #421), this is why I wrote it like that.